client/data/media-storage/with-media-storage.js

```diff
@@ -0,0 +1,16 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useSelector } from 'react-redux';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import useMediaStorageQuery from './use-media-storage-query';
+
+const withMediaStorage = createHigherOrderComponent(
+	( Wrapped ) => ( props ) => {
+		const siteId = useSelector( getSelectedSiteId );
+		const { data } = useMediaStorageQuery( siteId );
+
+		return <Wrapped { ...props } mediaStorage={ data ?? [] } />;
```

Comment:
`mediaStorage` seems to be an object when defined rather than an array. Shall we default to an empty object in this case?

Reply:
Not really sure how this ended up being an array. Likely a defect shift key 🤦 

Thanks for spotting! Fixed in 7e4799f73f9282dfd6130b60eaf98755b8fcb90e